components/MatterInterface/DoneCoffeeMaker.cpp

```diff
@@ -0,0 +1,539 @@
+    {
+        ESP_LOGI(TAG, "MicroSwitch Error");           
+        BuzzerPlay(BuzzerEffect_t::TRIPLE_BIZ);
+    }    
+    else 
+    {
+        if(valCurrentLevel.val.u8 == ON_MODE)
+        {
+
+        }
+    }
+}
+
+static app_driver_handle_t PowerKeyInit()
+{
+    app_driver_handle_t handle;
+    button_handle_t btnHandle;    
+
+    InitKeyWithPressCallback(btnHandle, 
+        CONFIG_DONE_COFFEE_MAKER_POWER_KEY,
+        PowerKeyCB);
+
+    iot_button_register_cb(btnHandle, 
+        BUTTON_LONG_PRESS_START, 
+        PowerKeyLongPressCB, NULL);
+    
+    InitKeyWithPressCallback(btnHandle, 
+        CONFIG_DONE_COFFEE_MAKER_MICRO_SWITCH,
+        MicroSwitchCB);
+
+    return (app_driver_handle_t)handle;
+}
+
+static void CookingModeGrindCB(void *Arg, void *Data)
+{
+    ESP_LOGI(TAG, "CookingModeGrindCB");
+    LevelControlUpdateCurrentValue(
+        INCREMENT_MODE, DONT_CARE, GrinderEndpointID);    
+    LevelControlUpdateCurrentValue(
+        EXPLICIT_MODE, GRINDER_MODE, CookingModeEndpointID);    
+}
+
+static void CookingModeCoffeeCB(void *Arg, void *Data)
+{
+    ESP_LOGI(TAG, "CookingModeCoffeeCB");
+    LevelControlUpdateCurrentValue(
+        EXPLICIT_MODE, COFFEE_MODE, CookingModeEndpointID);        
+}
+
+static void CookingModeTeaCB(void *Arg, void *Data)
+{
+    ESP_LOGI(TAG, "CookingModeTeaCB");
+    LevelControlUpdateCurrentValue(
+        EXPLICIT_MODE, TEA_MODE, CookingModeEndpointID);        
+}
+
+static app_driver_handle_t CookingModeInit()
+{
+    app_driver_handle_t handle;
+    button_handle_t btn1Handle, btn2Handle, btn3Handle;    
+
+    InitKeyWithPressCallback(btn1Handle, 
+        CONFIG_DONE_COFFEE_MAKER_COOKING_MODE_GRINDER,
+        CookingModeGrindCB);
+
+    InitKeyWithPressCallback(btn2Handle, 
+        CONFIG_DONE_COFFEE_MAKER_COOKING_MODE_COFFEE,
+        CookingModeCoffeeCB);    
+
+    InitKeyWithPressCallback(btn3Handle, 
+        CONFIG_DONE_COFFEE_MAKER_COOKING_MODE_TEA,
+        CookingModeTeaCB);        
+
+    return (app_driver_handle_t)handle;
+}
+
+static void CupCounterKeyCB(void *Arg, void *Data)
+{
+    ESP_LOGI(TAG, "CupCounterKeyCB");
+    LevelControlUpdateCurrentValue(
+        INCREMENT_MODE, DONT_CARE, CupCounterEndpointID);        
+}
+
+static app_driver_handle_t CupCounterInit()
+{
+    app_driver_handle_t handle;
+    button_handle_t btnHandle;    
+
+    InitKeyWithPressCallback(btnHandle, 
+        CONFIG_DONE_COFFEE_MAKER_CUP_COUNTER_KEY,
+        CupCounterKeyCB);
+
+    return (app_driver_handle_t)handle;
+}
+
+esp_err_t CookingModeSetEnable(
+    app_driver_handle_t DriverHandle, 
+    esp_matter_attr_val_t *Val)
+{
+    esp_err_t err = ESP_OK;
+
+    return err;
+}
+
+esp_err_t GrinderSetEnable(
+    app_driver_handle_t DriverHandle, 
+    esp_matter_attr_val_t *Val)
+{
+    esp_err_t err = ESP_OK;
+
+    return err;
+}
+
+esp_err_t CupCounterSetEnable(
+    app_driver_handle_t DriverHandle, 
+    esp_matter_attr_val_t *Val)
+{
+    esp_err_t err = ESP_OK;
+
```

Comment:
you never use error,just return ESP_OK !, use it some where or remove it.

Reply:
body of some callbacks not completed yet